File: app/src/main/java/com/pattern/mvvm/view/activity/MainActivity.java
package com.pattern.mvvm.view.activity;

import android.os.Bundle;
import androidx.lifecycle.ViewModelProviders;
import com.pattern.mvvm.BR;
import com.pattern.mvvm.R;
import com.pattern.mvvm.databinding.ActivityMainBinding;
import com.pattern.mvvm.viewmodel.MainViewModel;

public class MainActivity extends BaseActivity<ActivityMainBinding, MainViewModel> {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    MainViewModel getViewModel() {
        mViewModel = ViewModelProviders.of(this).get(MainViewModel.class);
        return mViewModel;
    }

    @Override
    int getBindingVariable() {
        return BR.main;
    }
}

File: app/src/main/java/com/pattern/mvvm/view/activity/MainActivity.kt
package com.pattern.mvvm.view.activity

import androidx.lifecycle.ViewModelProviders
import com.pattern.mvvm.BR
import com.pattern.mvvm.R
import com.pattern.mvvm.databinding.ActivityMainBinding
import com.pattern.mvvm.viewmodel.MainViewModel

class MainActivity : BaseActivity<ActivityMainBinding, MainViewModel>() {

    override val layoutId: Int = R.layout.activity_main

    override val getViewModel: MainViewModel
        get() {
            mViewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
            return mViewModel
        }

    override val bindingVariable: Int = BR.main

}
